language/borrow-graph/src/references/mod.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::paths::{self, Path};
+use crate::shared::*;
+use std::cmp::Ordering;
+use std::collections::{BTreeMap, BTreeSet};
+use std::fmt;
+use std::fmt::Debug;
+
+//**************************************************************************************************
+// Definitions
+//**************************************************************************************************
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub struct RefID(pub(crate) usize);
+
+impl RefID {
+    pub const fn new(x: usize) -> Self {
+        RefID(x)
+    }
+}
+
+#[derive(Clone)]
+pub(crate) struct BorrowEdge<Loc: Copy, Lbl: Clone + Ord> {
+    pub(crate) strong: bool,
```

Comment:
Did you consider making this an enum with two values?  This might be a matter of taste.  I don't have a strong opinion.

Reply:
I thought about it a bit, and I might update it be a enum, but it will be some work